sdk/tables/azure-data-tables/tests/test_retry.py

```diff
@@ -152,5 +200,113 @@ def test_no_retry(self, tables_storage_account_name, tables_primary_storage_acco
             self.ts.delete_table(new_table_name)
             self._tear_down()
 
+    @tables_decorator
+    @recorded_by_proxy
+    def test_failover_and_retry_on_secondary(self, tables_storage_account_name, tables_primary_storage_account_key):
+        url = self.account_url(tables_storage_account_name, "table")
+        table_name = self.get_resource_name("mytable")
+        entity = {"PartitionKey": "foo", "RowKey": "bar"}
+
+        # secondary endpoint only works on READ operations: get, list, query.
+        # retry request type in frozenset({'PUT', 'HEAD', 'TRACE', 'OPTIONS', 'DELETE', 'GET'})
+        with TableClient(
+            url,
+            table_name,
+            credential=tables_primary_storage_account_key,
+            retry_total=5,
+            retry_to_secondary=True,
+            transport=FailoverRetryTransport(),
+        ) as client:
+            with pytest.raises(ServiceRequestError) as ex:
+                client.create_table(failover=ServiceRequestError("Attempting to force failover"))  # POST, not retry
+            assert "Attempting to force failover" in str(ex.value)
+
+        with TableClient(
+            url,
+            table_name,
+            credential=tables_primary_storage_account_key,
+            retry_total=5,
+            retry_to_secondary=True,
+            transport=FailoverRetryTransport(),
+        ) as client:
+            client.create_table()
+            time.sleep(10)
+            # add the entity then run get_entity() again
+            client.create_entity(entity)
+            time.sleep(10)
+            client.get_entity(
+                "foo", "bar", failover=ServiceRequestError("Attempting to force failover")
```

Comment:
It would also be good to add some tests for error other than ServiceRequestError. For example:
- ServiceResponseError
- HttpResponseError (maybe test something like status code = 500)

Reply:
Added tests with `ServiceResponseError` and `HttpResponseError` separately with and without the enity ready on `get_entity()`